src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -902,6 +898,40 @@ partial class BoundConditionalAccess : IConditionalAccessExpression
         protected override OperationKind ExpressionKind => OperationKind.ConditionalAccessExpression;
     }
 
+    partial class BoundEqualsValue : ISymbolInitializer
+    {
+        IExpression ISymbolInitializer.Value => this.Value;
+
+        SyntaxNode IOperation.Syntax => this.Syntax;
+
+        bool IOperation.IsInvalid => ((IOperation)this.Value).IsInvalid;
+
+        OperationKind IOperation.Kind => this.OperationKind;
+
+        protected abstract OperationKind OperationKind { get; }
+    }
+
+    partial class BoundFieldEqualsValue : IFieldInitializer
+    {
+        IFieldSymbol IFieldInitializer.Field => this.Field;
+        
+        protected override OperationKind OperationKind => OperationKind.FieldInitializerAtDeclaration;
+    }
+
+    partial class BoundPropertyEqualsValue : IPropertyInitializer
+    {
+        IMethodSymbol IPropertyInitializer.Setter => this.Property?.SetMethod;
```

Comment:
`IPropertyInitializer` should probably expose `Property` rather than `Setter`. Otherwise, in an error case where the property is readonly, the analyzer will not be able to determine the property.